server/lib/parties/party-api.ts

```diff
@@ -112,19 +112,30 @@ export class PartyApi extends HttpApi {
       )
       .delete(
         '/:partyId/:clientId',
-        throttleMiddleware(partyThrottle, ctx => String(ctx.session!.userId)),
-        leave,
+        async (ctx, next) => {
+          let throttle
+          if (ctx.query.type === 'leave') {
```

Comment:
Maybe just put this logic in the leaveOrKick method, after validations run? (I also question whether we really need a separate kick and party action throttle at all, seems kinda unnecessarily granular for how inexpensive the requests are)

side note: you can also use the `apiEndpoint` helper I wrote to make including the validations easier